### PR TITLE
Some post-rework build/compile fixes

### PR DESCRIPTION
**Pull request by TheComputerGuy96**

### CODE REVIEW

**Comment by mikelow** [comment]
Can you state what OS and compiler you had difficulties with?

**Comment by TheComputerGuy96** [comment]
> Can you state what OS and compiler you had difficulties with?

I'm using Arch Linux with GCC 13.2.1 :frog:

**Comment by mikelow** [comment]
What is the issue you are having with the git submodule relative path? That wasn't changed by the refactor merge.

**Comment by TheComputerGuy96** [comment]
> What is the issue you are having with the git submodule relative path? That wasn't changed by the refactor merge.

That's an issue I found when trying to configure spdlog submodule (which was added as a dependency in the rework); I'm surprised no one catched it before

**Comment by mikelow** [comment]
> That's an issue I found when trying to configure spdlog submodule (which was added as a dependency in the rework); I'm surprised no one catched it before

What is the error you are experiencing? And may as well ask, what version of git?

File: .gitmodules
Comment (sykhro) [inline .gitmodules:1]: What is the issue here? If anything, the real problem can be the spdlog url

**Review by sykhro — CHANGES_REQUESTED** [review]

**Comment by TheComputerGuy96** [comment]
> What is the error you are experiencing? And may as well ask, what version of git?

Here's the error I have with `git submodule update --init` (with git 2.45.0):
```
❯ git submodule update --init
fatal: repository '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' does not exist  
fatal: clone of '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' into submodule path '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/src/vgmtrans/lib/vgmtrans-ext-win' failed  
Failed to clone 'lib/vgmtrans-ext-win'. Retry scheduled  
fatal: repository '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' does not exist  
fatal: clone of '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' into submodule path '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/src/vgmtrans/lib/vgmtrans-ext-win' failed  
Failed to clone 'lib/vgmtrans-ext-win' a second time, aborting  
```

File: .gitmodules
Comment (TheComputerGuy96) [reply .gitmodules:1]: I just posted a comment showing the exact issue I have (I think my fix is correct)

**Comment by sykhro** [comment]
> > What is the error you are experiencing? And may as well ask, what version of git?
> 
> Here's the error I have with `git submodule update --init` (with git 2.45.0):
> 
> ```
> ❯ git submodule update --init
> fatal: repository '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' does not exist  
> fatal: clone of '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' into submodule path '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/src/vgmtrans/lib/vgmtrans-ext-win' failed  
> Failed to clone 'lib/vgmtrans-ext-win'. Retry scheduled  
> fatal: repository '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' does not exist  
> fatal: clone of '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/vgmtrans-ext-win' into submodule path '/mnt/storagebeast/Development/AUR/Published/vgmtrans-git/src/vgmtrans/lib/vgmtrans-ext-win' failed  
> Failed to clone 'lib/vgmtrans-ext-win' a second time, aborting  
> ```

you need `git submodule init` first, so that git can resolve the right absolute url. relative submodule paths ensure that git picks https or ssh depending on what was used to clone the main repository. defaulting to either can break people workflows (especially in case of a private fork)

**Comment by TheComputerGuy96** [comment]
> you need `git submodule init` first, so that git can resolve the right absolute url.

I already ran `git submodule init` before and I still have the same issue

**Comment by TheComputerGuy96** [comment]
Any updates on the PR? Should I split the submodule change to a separate PR (because it's somehow controversial)?

**Comment by sykhro** [comment]
sorry, i missed your reply. as previously stated, the https endpoint can break the workflow of people using git with ssh. relative submodule paths are common practice to avoid this problem.

did you fork the repo via github and then clone? in that case you will also need to fork the submodule.
to avoid this issue, clone this repo first and push to your fork remote afterwards

**Comment by TheComputerGuy96** [comment]
I dropped the submodule change (apparently my problem is caused by makepkg using two separate .git folders for each repository and the one with all the files using a local URL for the git remote which links to that other one with all the objects) :frog:

**Review by sykhro — APPROVED** [review]

**Comment by sykhro** [comment]
Thanks for your contribution!

**Pull request merged**